tests_e2e/tests/lib/agent_update_helpers.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+
+import requests
+from azure.identity import DefaultAzureCredential
+from msrestazure.azure_cloud import Cloud
+from azure.mgmt.compute.models import VirtualMachine
+
+from tests_e2e.tests.lib.azure_clouds import AZURE_CLOUDS
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.virtual_machine_client import VirtualMachineClient
+
+# Helper methods for agent update/publish tests
+
+
+def verify_agent_update_flag_enabled(vm: VirtualMachineClient) -> bool:
+    result: VirtualMachine = vm.get_model()
+    flag: bool = result.os_profile.linux_configuration.enable_vm_agent_platform_updates
+    if flag is None:
+        return False
+    return flag
+
+
+def enable_agent_update_flag(vm: VirtualMachineClient) -> None:
+    osprofile = {
+        "location": vm.location,  # location is required field
+        "properties": {
+            "osProfile": {
+                "linuxConfiguration": {
+                    "enableVMAgentPlatformUpdates": True
+                }
+            }
+        }
+    }
+    log.info("updating the vm with osProfile property:\n%s", osprofile)
+    vm.update(osprofile)
+
+
+def request_rsm_update(requested_version: str, vm: VirtualMachineClient) -> None:
+    """
+    This method is to simulate the rsm request.
+    First we ensure the PlatformUpdates enabled in the vm and then make a request using rest api
+    """
+    if not verify_agent_update_flag_enabled(vm):
+        # enable the flag
+        log.info("Attempting vm update to set the enableVMAgentPlatformUpdates flag")
+        enable_agent_update_flag(vm)
```

Comment:
Why is this necessary? How is it used vs. our waagent.conf flag

Reply:
This different flag need to set in the vm model for CRP/RSM driving RSM updates